stubs/pyserial/serial/tools/miniterm.pyi

```diff
@@ -3,16 +3,27 @@ import sys
 import threading
 from _typeshed import Unused
 from collections.abc import Iterable
-from typing import Any, BinaryIO, TextIO
+from typing import Any, Generic, Protocol, TypeVar, type_check_only
 from typing_extensions import Self
 
 from serial import Serial
 
+_AnyStr_T = TypeVar("_AnyStr_T", contravariant=True)
+
+@type_check_only
+class _Writer(Protocol, Generic[_AnyStr_T]):
+    def write(self, s: _AnyStr_T, /) -> Unused: ...
+    def flush(self) -> Unused: ...
+
+@type_check_only
+class _Reader(Protocol):
+    def read(self, n: int, /) -> str: ...
```

Comment:
Unfortunate this doesn't match `SupportsRead` :/ (optional parameter in the existing protocol)